tests/std/tests/GH_005421_vector_algorithms_integer_class_type_iterator/test.cpp

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <__msvc_int128.hpp>
+#include <algorithm>
+#include <cassert>
+#include <compare>
+#include <iterator>
+
+#include "range_algorithm_support.hpp"
+
+using namespace std;
+
+template <class T>
+using picky_contiguous_iterator = test::redifference_iterator<_Signed128, T*>;

```

Comment:
In a previous version of this PR, custom iterator was implemented here. 
If there's need to make this tests into pre-C++20 modes, it can be restored as a starting point for that.

Although I do not want to implement this test for pre-C++20 modes.

Reply:
I think pre-C++20 iterators are not allowed to use non-integer types as their difference types, so this test seems invalid for pre-C++20 modes.

Reply:
Didn't know that. But then I think with an unusual difference type (like `short`) there would be the same casts needed.

Reply:
Having the test be usual_20_matrix seems fine, and I agree that this should additionally help defend against narrow difference types in older modes without explicit coverage of that.

Reply:
> narrow difference types in older modes 

Just realized that this seems impossible. There's no `contiguous_iterator` in older modes, so it should be pointers with custom `iterator_traits`. But then it should be pointers to user defined types. Which we can't vectorize because don't know if they compare bitwise.